dotCMS/src/main/java/com/dotmarketing/portlets/workflows/business/WorkflowAPIImpl.java

```diff
@@ -220,6 +221,38 @@ public void saveSchemesForStruct(final Structure struc, final List<WorkflowSchem
 		}
 	}
 
+	@CloseDBIfOpened
+	@Override
+	public List<WorkflowScheme> findSchemesForContentType(final ContentType contentType) throws DotDataException {
+
+		final ImmutableList.Builder<WorkflowScheme> schemes =
+				new ImmutableList.Builder<>();
+
+		if (contentType == null || ! UtilMethods.isSet(contentType.inode())
+				|| LicenseUtil.getLevel() < LicenseLevel.STANDARD.level) {
+
+			schemes.add(findDefaultScheme());
+		} else {
+
+			try {
+
+				Logger.debug(this, "Finding the schemes for: " + contentType);
+				final List<WorkflowScheme> contentTypeSchemes =
+						this.workFlowFactory.findSchemesForStruct(contentType.inode());
+				if(contentTypeSchemes.isEmpty()){
+					schemes.add(findDefaultScheme());
+				} else {
+					schemes.addAll(contentTypeSchemes);
+				}
+			}
+			catch(Exception e) {

```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1166)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Move this "catch" on the same line that the previous closing curly brace. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARightCurlyBraceSameLineAsNextBlockCheck)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Catch a list of specific exception subtypes instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2221)

Reply:
Not needed